tests/test_transfer.py

```diff
@@ -247,6 +249,98 @@ def test_disk_vs_tape_with_custom_strategy(rse_factory, root_account, mock_scope
         assert transfer[0].src.rse.name == tape_rse_name
 
 
+@pytest.mark.parametrize("file_config_mock", [
+    {"overrides": [('transfers', 'source_ranking_strategies', 'PathDistance')]},
+    {"overrides": [('transfers', 'source_ranking_strategies', 'TransferWaitTime,PathDistance')]}
+], indirect=True)
+def test_wait_time_with_custom_strategy(rse_factory, root_account, mock_scope, file_config_mock):
+    """
+    Sources that have shorter wait times are more likely to be picked if the TransferWaitTime strategy is set.
+    """
+    shortest_wait_rse_name, shortest_wait_rse_id = rse_factory.make_posix_rse()
+    short_wait_rse_name, short_wait_rse_id = rse_factory.make_posix_rse()
+    long_wait_rse_name, long_wait_rse_id = rse_factory.make_posix_rse()
+    longest_wait_rse_name, longest_wait_rse_id = rse_factory.make_posix_rse()
+    dst_rse_name, dst_rse_id = rse_factory.make_posix_rse()
+    all_rses = [shortest_wait_rse_id, short_wait_rse_id, long_wait_rse_id, longest_wait_rse_id, dst_rse_id]
+    add_distance(shortest_wait_rse_id, dst_rse_id, distance=40)
+    add_distance(short_wait_rse_id, dst_rse_id, distance=30)
+    add_distance(long_wait_rse_id, dst_rse_id, distance=20)
+    add_distance(longest_wait_rse_id, dst_rse_id, distance=10)
+
+    # Add mock data about existing source queues
+    db_session = get_session()
+
```

Comment:
just decorate the function with the transactional_session decorator. Which will take care of creating the session and commiting it.